pkg/reconciler/run/controller.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package run
+
+import (
+	"context"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline"
+	pipelineclient "github.com/tektoncd/pipeline/pkg/client/injection/client"
+	runinformer "github.com/tektoncd/pipeline/pkg/client/injection/informers/pipeline/v1alpha1/run"
+	runreconciler "github.com/tektoncd/pipeline/pkg/client/injection/reconciler/pipeline/v1alpha1/run"
+	"k8s.io/client-go/tools/cache"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/tracker"
+)
+
+// NewController instantiates a new controller.Impl from knative.dev/pkg/controller
+func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	logger := logging.FromContext(ctx)
+	pipelineclientset := pipelineclient.Get(ctx)
+	runInformer := runinformer.Get(ctx)
+
+	c := &Reconciler{
+		pipelineClientSet: pipelineclientset,
+	}
+	impl := runreconciler.NewImpl(ctx, c, func(impl *controller.Impl) controller.Options {
+		configStore := config.NewStore(logger.Named("config-store"))
+		configStore.WatchConfigs(cmw)
+
+		return controller.Options{
+			AgentName:   pipeline.RunControllerName,
+			ConfigStore: configStore,
+		}
+	})
+
+	logger.Info("Setting up event handlers")
+	runInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    impl.Enqueue,
+		UpdateFunc: controller.PassNew(impl.Enqueue),
+	})
+	c.tracker = tracker.New(impl.EnqueueKey, controller.GetTrackerLease(ctx))
+
+	return impl
+}
```

Comment:
if the intention of this controller is for folks to take it and build on it (I think? side note im not quite sure what this controller is for from the PR description / commit message) then maybe it would make sense to include some comments for every few lines explaining whats going on?